macroslib/src/cpp/mod.rs

```diff
@@ -517,21 +564,12 @@ public:
     let mut gen_code = Vec::new();
 
     let (this_type_for_method, code_box_this) =
-        if let (Some(this_type), Some(constructor_ret_type)) = (
+        if let (Some(this_type), Some(_constructor_ret_type)) = (
```

Comment:
If there is no need for `_constructor_ret_type` why bother to take it from expression?

Reply:
The code mimics old behaviour: only do `add_type` for `this_type_for_method` and `constructor_ret_type`, and generate `impl SwigForeignClass`, if both `this_type_for_method` and `constructor_ret_type` are present.

Now, the type registration and code generation is separated, but the behaviour is the same: if `constructor_ret_type` is `None`, code for `impl SwigForeignClass` won't be generated.

That being said, the relationship between `self_type`, `this_type_for_method`, `constructor_ret_type` and `SwigForeignClass` is a bit vague for me, which is why I decided to just recreate old behaviour, without digging into it.